IntelliTect.TestTools.TestFramework.Tests/TestDataTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using Xunit;
+
+namespace IntelliTect.TestTools.TestFramework.Tests;
+
+public class TestDataTests
+{
+    [Fact]
+    public void BlockDataT1T2ThrowsExceptionOnDuplicateTypes()
+    {
+        var exception = Assert.Throws<TypeInitializationException>(() => 
+        {
+            var _ = new BlockData<int, int>(1, 2);
+        });
+        Assert.NotNull(exception.InnerException);
+        Assert.Equal(typeof(InvalidOperationException), exception.InnerException.GetType());
+        Assert.Equal("Duplicate type found: Int32 appears multiple times. BlockData must use different types to avoid unexpected behavior by the TestCase DI Container.",
+            exception.InnerException.Message);
+    }
+
+    [Fact]
+    public void BlockDataT1T2T3ThrowsExceptionOnDuplicateTypes()
+    {
+        var exception = Assert.Throws<TypeInitializationException>(() =>
+        {
+            var _ = new BlockData<int, bool, int>(1, true, 2);
+        });
+        Assert.NotNull(exception.InnerException);
+        Assert.Equal(typeof(InvalidOperationException), exception.InnerException.GetType());
+        Assert.Equal("Duplicate type found: Int32 appears multiple times. BlockData must use different types to avoid unexpected behavior by the TestCase DI Container.",
+            exception.InnerException.Message);
```

Comment:
```suggestion
        var invalidOp = Assert.IsType<InvalidOperationException>(exception.InnerException);
        Assert.Equal("Duplicate type found: Int32 appears multiple times. BlockData must use different types to avoid unexpected behavior by the TestCase DI Container.",
            invalidOp.Message);
```